### PR TITLE
adding function to emacs editor integration to get project root.

### PR DESCRIPTION
**Pull request by yilinwei**
Add a helper function for user scripts to get the root directory of the current dune project.

### CODE REVIEW

**Comment by Alizter** [comment]
This won't give the correct project root since you can have a dune-project file in a subdirectory with respect to another. The outermost dune project file would end up being the root of the workspace. And if there is a dune-workspace outside of that, then that will be the root.

Of course, this is how I understood the code.

**Comment by yilinwei** [comment]
Thanks @Alizter; I didn't realise that the `dune-project`/`dune-workspace` could be recursive. I have corrected it according to the [docs](https://dune.readthedocs.io/en/stable/usage.html#finding-the-root) and done a quick test on my system with a test project.

File: editor-integration/emacs/dune.el [hunk 1]
```diff
@@ -424,6 +424,28 @@ For customization purposes, use `dune-mode-hook'."
...
+DIRECTORY defaults to `default-directory' if not provided."
+  (let
+      (root
+       workspace
+       (dir (or directory default-directory)))
+    (while dir
+      (cond
+       ((file-exists-p
+	 (file-name-concat dir "dune-workspace"))
+	(setq workspace t
```
Comment (anmonteiro) [inline editor-integration/emacs/dune.el:439]: what's `t`?

**Review by anmonteiro — APPROVED** [review]

**Comment by rgrinberg** [comment]
@Gopiandcode could you take a look at this PR?

**Review by Alizter — APPROVED** [review]
@ejgallego might be able to comment too.

**Comment by ejgallego** [comment]
Looks OK to me, not an expert on Emacs tho, only a user :hammer_and_pick:

**Comment by Gopiandcode** [comment]
Yes, looks good to me as well.

**Comment by yilinwei** [comment]
@Gopiandcode 

Thanks! Should have expected that Emacs had a utility function for file traversals.

**Pull request merged**